WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -86,105 +121,166 @@ private extension ReviewsDashboardCardViewModel {
+        }
+
+        do {
+            try notificationsResultsController.performFetch()
+            updateData()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
     }
 }
 
 // MARK: - Private helpers
 private extension ReviewsDashboardCardViewModel {
-    @MainActor
-    func loadReviews() async throws {
-        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
-            stores.dispatch(ProductReviewAction.synchronizeProductReviews(siteID: siteID,
-                                                                          pageNumber: 1,
-                                                                          pageSize: Constants.numberOfItems) { result in
-                switch result {
-                case .success:
-                    // Ignoring the result from remote as we're using storage as the single source of truth
-                    continuation.resume()
-                case .failure(let error):
-                    continuation.resume(throwing: error)
+    /// Populates data from the local storage for the three main data (reviews, products, notifications).
+    ///
+    func updateData() {
+        var filteredLocalReviews: [ProductReview] = []
+        if currentFilter != .all {
+            filteredLocalReviews = reviews.filter { $0.status == currentFilter.productReviewStatus }
+        } else {
+            filteredLocalReviews = reviews
+        }
+
+        // Ensure to only get the first three items.
+        let localReviews = filteredLocalReviews.prefix(Constants.numberOfItems)
+
+        data = localReviews.map { review in
+
+            // Depending on the sync progress, `product` and `notification` might still be nil.
+            // This is acceptable and the app is able to display partial review content.
+            let product = reviewProducts.first { $0.productID == review.productID }
+            let notification = notifications.first { notification in
+                if let notificationReviewID = notification.meta.identifier(forKey: .comment) {
+                    return notificationReviewID == review.reviewID
                 }
-            })
+                return false
+            }
+            return ReviewViewModel(
+                showProductTitle: product != nil,
+                review: review,
+                product: product,
+                notification: notification
+            )
         }
     }
 
-    /// Updates data
     @MainActor
-    func updateReviewsResults() async {
-        let reviews = productReviewsResultsController.fetchedObjects.prefix(Constants.numberOfItems)
-        let productIDs = reviews.map { $0.productID }
+    func synchronizeReviews(filter: ReviewsFilter) async throws {
+        let fetchedReviews = try await synchronizeProductReviews(filter: filter)
+
+        let productIDs = fetchedReviews.prefix(Constants.numberOfItems).map { $0.productID }
```

Comment:
Nit: using `prefix` is not necessary because you already defined the number of items in `synchronizeProductReviews` request.

Reply:
You're right as this from a network request. Updated in https://github.com/woocommerce/woocommerce-ios/pull/12769/commits/595c389cf45c1e207c5912f221b5173b79b6a692